file.h

```diff
@@ -12,6 +12,10 @@ class File
 public:
     File(string name,string content = ""):
         name(name),content(content){}
+
+    bool operator ==(File* f){
+        return f->name == this->name;
+    }
```

Comment:
ez baromi csunya igy amugy , hogy pointerre meg adatra is megcsinalod. Inkabb csak adatra, es ahol pointered van az == hasznalatanal, ott  tedd ele a *-ot.